test/test.sh

```diff
@@ -141,25 +141,40 @@ function shouldStartEntity() {
     waitForCommand isEntityStatus "${appname}" "${entityname}" RUNNING || fail; return
 }
 
+function usage() {
+cat <<EOU
+Usage: $0 [-h] [-a <application_filename> ] <brooklyn_url> [ <user> <password> ]
+-h  show this help
+-a  deploy the file named <application_filename> instead of the default test_app.yaml
+EOU
+exit 0
+}
+
+
 
 function main() {
 
     local application_yaml=test_app.yaml
     local appname=$(grep name: ${application_yaml} | sed 's/.*: *//')
```

Comment:
A YAML file may have multiple lines that match 'name: '.  Suggest that you match on the first one found with:
`grep name: ${application_yaml} | head -1 | sed 's/.*: *//'`
Or possibly retrieve the name from the output of the deploy command?